src/propTypes/albumTypes.js

```diff
@@ -0,0 +1,28 @@
+import PropTypes from 'prop-types';
+
+export const albumShape = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
+  albumName: PropTypes.string.isRequired,
+  artistName: PropTypes.string.isRequired,
+  image: PropTypes.string.isRequired,
+  // custom validator function takes three arguments
+  releaseDate: (props, propName, componentName) => {
+    if (!/^\d{4}-\d{2}-\d{2}$/.test(props[propName])) {
+      // format YYYY-MM-DD
+      return new Error(
+        `Invalid prop ${propName} supplied to ${componentName}. Validation failed.`
+      );
+    }
+  },
+  averageRating: (props, propName, componentName) => {
+    if (
+      typeof props[propName] !== 'number' ||
+      props[propName] < 1 ||
+      props[propName] > 5
+    ) {
+      return new Error(
+        `Invalid prop ${propName} supplied to ${componentName}. Validation failed.`
```

Comment:
Not a big deal, but for both of these custom props it might be nice to have a bit more info in the error returned, about _why_ validation failed. For example for the rating, we could say "Validation failed. averageRating must be a number between 0 and 5". 

Makes it a bit easier for another developer to know what's gone wrong, without having to go and open up this file to inspect the exact validation code.